lib/tests/streamlit/watcher/LocalSourcesWatcher_test.py

```diff
@@ -77,6 +77,18 @@ def test_just_script(self, fob):
 
         self.assertEqual(fob.call_count, 1)  # __init__.py
 
+    @patch("streamlit.watcher.LocalSourcesWatcher.FileWatcher")
+    def test_permission_error(self, fob):
+        from streamlit import compatibility
+
+        if compatibility.is_running_py3():
+            ErrorType = PermissionError
+        else:
+            ErrorType = OSError
+
+        fob.side_effect = ErrorType('This error should be caught!')
+        lso = LocalSourcesWatcher.LocalSourcesWatcher(REPORT, NOOP_CALLBACK)
```

Comment:
We could add a check that `lso` is actually set up correctly, like for example, we can test that `_register_watcher` has been called.

Reply:
Hmmm... We shouldn't check that internal functions are called. That feels like testing the implementation rather than the behavior.

We already test `LocalSourcesWatcher` quite a bit in the rest of this file. For example, in the test below this one we check that `lso` actually triggers the callback when a watched file changes. So I think this is well tested as is.

Reply:
Sounds good to me.